homeassistant/components/stream/worker.py

```diff
@@ -64,11 +64,16 @@ def _stream_worker_internal(hass, stream, quit_event):
         video_stream = container.streams.video[0]
     except (KeyError, IndexError):
         _LOGGER.error("Stream has no video")
+        container.close()
         return
     try:
         audio_stream = container.streams.audio[0]
     except (KeyError, IndexError):
         audio_stream = None
+    # These formats need aac_adtstoasc bitstream filter, but auto_bsf not
+    # compatible with empty_moov and manual bitstream filters not in PyAV
+    if container.format.name in {"hls", "mpegts"}:
```

Comment:
This is probably the easiest way to do it. I am not sure if "hls" and "mpegts" are the names that are returned for those inputs. I do know that "rtsp" is what is returned for a rtsp input. We could change the check to (container.format.name != "rtsp") instead.
Also, we could just disable the output audio stream for fmp4 instead of disabling the input audio stream. This would allow the recorder to still get audio.

Reply:
I think the recorder will probably have the same problem (muxing to mp4). This is fine for now, assuming it works :)

Reply:
Actually I don't think it would because we aren't setting the empty_moov flag on the recorder segments, so the bitstream filter would be automatically applied. But the lookback segments come from the hls segments, and those would have no audio stream...messy.
I agree best to keep it simple especially if it's not a long term solution anyway.